packages/console/src/components/RoleScopesTransfer/components/SourceScopeItem/index.tsx

```diff
@@ -0,0 +1,39 @@
+import type { ScopeResponse } from '@logto/schemas';
+
+import Checkbox from '@/components/Checkbox';
+import { onKeyDownHandler } from '@/utilities/a11y';
+
+import * as styles from './index.module.scss';
+
+type Props = {
+  scope: ScopeResponse;
+  isSelected: boolean;
+  onSelect: (scope: ScopeResponse) => void;
+};
+
+const SourceScopeItem = ({ scope, scope: { name }, isSelected, onSelect }: Props) => (
+  <div className={styles.sourceScopeItem}>
+    <Checkbox
+      checked={isSelected}
+      disabled={false}
+      onChange={() => {
+        onSelect(scope);
+      }}
+    />
+    <div
+      className={styles.name}
+      role="button"
+      tabIndex={0}
+      onKeyDown={onKeyDownHandler(() => {
+        onSelect(scope);
+      })}
+      onClick={() => {
+        onSelect(scope);
+      }}
+    >
+      {name}
+    </div>
+  </div>
+);
+
+export default SourceScopeItem;
```

Comment:
Can the `SourceScopeItem` and `SourceUserItem` be extracted into a shared component? 
Same thing for `SourceScopesBox` and `SourceUsersBox`

Reply:
I tried, but I found it hard to do that since they have different layout limits and logic, and the shared component will be complex in logic and styles.